launchpad/src/setup.rs

```diff
@@ -0,0 +1,207 @@
+    }
+
+    #[only_owner]
+    #[endpoint(setWinnerSelectionStartEpoch)]
+    fn set_winner_selection_start_epoch(&self, start_epoch: u64) -> SCResult<()> {
+        self.try_set_winner_selection_start_epoch(start_epoch)
+    }
+
+    #[only_owner]
+    #[endpoint]
+    fn set_confirmation_period_in_epochs(&self, confirmation_period: u64) -> SCResult<()> {
+        self.try_set_confirmation_period_in_epochs(confirmation_period)
+    }
+
+    #[only_owner]
+    #[endpoint(setClaimPeriodInEpochs)]
+    fn set_claim_period_in_epochs(&self, claim_period: u64) -> SCResult<()> {
+        self.try_set_claim_period_in_epochs(claim_period)
+    }
+
+    #[only_owner]
+    #[endpoint(setTicketPaymentToken)]
+    fn set_ticket_payment_token(&self, ticket_payment_token: TokenIdentifier) -> SCResult<()> {
+        self.try_set_ticket_payment_token(&ticket_payment_token)
+    }
+
+    #[only_owner]
+    #[endpoint]
+    fn set_ticket_price(&self, ticket_price: Self::BigUint) -> SCResult<()> {
+        self.try_set_ticket_price(&ticket_price)
+    }
+
+    #[only_owner]
+    #[endpoint]
+    fn set_launchpad_tokens_per_winning_ticket(&self, amount: Self::BigUint) -> SCResult<()> {
+        require!(
+            amount > 0,
+            "Launchpad tokens per winning ticket cannot be set to zero"
+        );
+
+        self.launchpad_tokens_per_winning_ticket().set(&amount);
+
+        Ok(())
+    }
+
+    #[only_owner]
+    #[endpoint(addAddressToBlacklist)]
+    fn add_address_to_blacklist(&self, address: Address) -> SCResult<()> {
+        self.blacklist().insert(address);
+
+        Ok(())
+    }
+
+    #[only_owner]
+    #[endpoint(removeAddressFromBlacklist)]
+    fn remove_address_from_blacklist(&self, address: Address) -> SCResult<()> {
+        self.blacklist().remove(&address);
+
+        Ok(())
+    }
+
+    // private
+
+    fn try_set_ticket_payment_token(&self, ticket_payment_token: &TokenIdentifier) -> SCResult<()> {
+        require!(
+            ticket_payment_token.is_egld() || ticket_payment_token.is_valid_esdt_identifier(),
+            "Invalid ticket payment token"
+        );
+
+        self.ticket_payment_token().set(ticket_payment_token);
+
+        Ok(())
+    }
+
+    fn try_set_ticket_price(&self, ticket_price: &Self::BigUint) -> SCResult<()> {
+        require!(ticket_price > &0, "Ticket price must be higher than 0");
+
+        self.ticket_price().set(ticket_price);
+
+        Ok(())
+    }
+
+    fn try_set_nr_winning_tickets(&self, nr_winning_tickets: usize) -> SCResult<()> {
+        require!(nr_winning_tickets > 0, "Cannot set number of winning tickets to zero");
+
+        self.nr_winning_tickets().set(&nr_winning_tickets);
+
+        Ok(())
+    }
+
+    fn try_set_winner_selection_start_epoch(&self, start_epoch: u64) -> SCResult<()> {
+        let current_epoch = self.blockchain().get_block_epoch();
+        require!(
+            start_epoch > current_epoch,
+            "Start epoch cannot be in the past"
+        );
+
```

Comment:
add more require in these sets - to verify that they are in the specified order. selectWinners start end, confirm ticket start and end, start for selling launchpad tokens